File: lib_base/src/main/java/com/android/base/app/fragment/delegates/FragmentDelegate.java
package com.android.base.app.fragment.delegates;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public interface FragmentDelegate<T extends Fragment> {

    /**
     * 当该Delegate被添加到Fragment中
     */
    default void onAttachToFragment(T fragment) {
    }

    /**
     * 调用此方法时，清除Fragment的引用
     */
    default void onDetachFromFragment() {
    }

    default void onAttach(Context context) {
    }

    default void onCreate(@Nullable Bundle savedInstanceState) {
    }

    default void onActivityCreated(@Nullable Bundle savedInstanceState) {
    }

    default void onSaveInstanceState(Bundle savedInstanceState) {
    }

    default void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
    }

    default void onStart() {
    }

    default void onResume() {
    }

    default void onPause() {
    }

    default void onStop() {
    }

    default void onDestroy() {
    }

    default void onDestroyView() {
    }

    default void onDetach() {
    }

    default void setUserVisibleHint(boolean isVisibleToUser) {
    }

    default void onHiddenChanged(boolean hidden) {
    }

    default void onActivityResult(int requestCode, int resultCode, Intent data) {
    }

    default void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
    }

}

File: lib_foundation/src/main/java/com/android/base/foundation/fragment/FragmentDelegate.kt
package com.android.base.foundation.fragment

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.annotation.UiThread
import androidx.fragment.app.Fragment

@UiThread
interface FragmentDelegate<T : Fragment> {

    /**该Delegate被添加到Fragment中*/
    fun onAttachToFragment(fragment: T) {}

    /**调用此方法时，清除Fragment的引用*/
    fun onDetachFromFragment() {}

    fun onAttach(context: Context) {}
    fun onCreate(savedInstanceState: Bundle?) {}
    fun onActivityCreated(savedInstanceState: Bundle?) {}
    fun onSaveInstanceState(savedInstanceState: Bundle) {}
    fun onViewCreated(view: View, savedInstanceState: Bundle?) {}
    fun onStart() {}
    fun onResume() {}
    fun onPause() {}
    fun onStop() {}
    fun onDestroy() {}
    fun onDestroyView() {}
    fun onDetach() {}

    fun setUserVisibleHint(isVisibleToUser: Boolean) {}
    fun onHiddenChanged(hidden: Boolean) {}

    fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {}
    fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {}

}